src/EnergyPlus/DaylightingManager.cc

```diff
@@ -4371,543 +4345,663 @@ namespace DaylightingManager {
+					ErrorsFound = true;
+					continue;
+				}
+			} else if (zone_daylight.DaylightMethod == SplitFluxDaylighting) {
+				ShowWarningError( "No " + cAlphaFieldNames( 6 ) + " provided for object named: " + cAlphaArgs( 1 ) );
+				ShowContinueError( "No glare calculation performed, and the simulation continues." );
 			}
-			zone_daylight.DaylightType = DetailedDaylighting;
-			zone_daylight.TotalDaylRefPoints = rNumericArgs( 1 );
 
-			rLightLevel = GetDesignLightingLevelForZone( ZoneFound );
-			CheckLightsReplaceableMinMaxForZone( ZoneFound );
+			zone_daylight.ViewAzimuthForGlare = rNumericArgs( 5 ); // Field: Glare Calculation Azimuth Angle of View Direction Clockwise from Zone y-Axis
+			zone_daylight.MaxGlareallowed = rNumericArgs( 6 ); // Field: Maximum Allowable Discomfort Glare Index
+			zone_daylight.DElightGriddingResolution = rNumericArgs( 7 ); // Field: DElight Gridding Resolution
 
-			zone_daylight.DaylRefPtAbsCoord.allocate( 3, MaxRefPoints );
-			zone_daylight.DaylRefPtAbsCoord = 0.0;
-			zone_daylight.DaylRefPtInBounds.allocate( MaxRefPoints );
-			zone_daylight.DaylRefPtInBounds = true;
-			zone_daylight.FracZoneDaylit.allocate( MaxRefPoints );
+			int curTotalDaylRefPts = NumAlpha - 6; // first six alpha fields are not part of extensible group
+			zone_daylight.TotalDaylRefPoints = curTotalDaylRefPts;
+			if ( ( NumNumber - 7 ) / 2 != zone_daylight.TotalDaylRefPoints ){
+				ShowSevereError( cCurrentModuleObject + "The number of extensible numeric fields and alpha fields is inconsistent for: " + cAlphaArgs( 1 ) );
+				ShowContinueError( "For each field: " + cAlphaFieldNames( NumAlpha ) + " there needs to be the following fields: Fraction Controlled by Reference Point and Illuminance Setpoint at Reference Point" );
+				ErrorsFound = true;
+			}
+			zone_daylight.DaylRefPtNum.allocate( curTotalDaylRefPts );
+			zone_daylight.DaylRefPtNum = 0;
+			zone_daylight.FracZoneDaylit.allocate( curTotalDaylRefPts );
 			zone_daylight.FracZoneDaylit = 0.0;
-			zone_daylight.IllumSetPoint.allocate( MaxRefPoints );
+			zone_daylight.IllumSetPoint.allocate( curTotalDaylRefPts );
 			zone_daylight.IllumSetPoint = 0.0;
-			zone_daylight.RefPtPowerReductionFactor.allocate( MaxRefPoints );
-			zone_daylight.RefPtPowerReductionFactor = 1.0;
-			zone_daylight.DaylIllumAtRefPt.allocate( MaxRefPoints );
+			zone_daylight.DaylIllumAtRefPt.allocate( curTotalDaylRefPts );
 			zone_daylight.DaylIllumAtRefPt = 0.0;
-			zone_daylight.GlareIndexAtRefPt.allocate( MaxRefPoints );
+			zone_daylight.GlareIndexAtRefPt.allocate( curTotalDaylRefPts );
 			zone_daylight.GlareIndexAtRefPt = 0.0;
-			zone_daylight.BacLum.allocate( MaxRefPoints );
+
+			zone_daylight.DaylRefPtAbsCoord.allocate( 3, curTotalDaylRefPts );
+			zone_daylight.DaylRefPtAbsCoord = 0.0;
+			zone_daylight.DaylRefPtInBounds.allocate( curTotalDaylRefPts );
+			zone_daylight.DaylRefPtInBounds = true;
+			zone_daylight.RefPtPowerReductionFactor.allocate( curTotalDaylRefPts );
```

Comment:
All these variables are allocated to the same amount, and they all appear to be simple int/real/bool variables.  Why isn't there just a single struct:

```
struct DaylightVars {
  int daylRefPtNum;
  Real64 fracZoneDayLit;
  //etc...
}
```

That is then assigned to the daylight structure or whatever and allocated in one line...  

I get that maybe that's just out of scope for here.  So maybe I'm just noting it for the sake of hearing my keyboard typing...